src/Storages/MergeTree/GinIndexStore.h

```diff
@@ -157,6 +205,11 @@ class GinIndexStore
 
     UInt32 getCurrentSegmentID() const { return current_segment.segment_id; }
 
+    /// Set the unique value count, needed to size the bloom filter.
+    /// TODO: The unique row count is actually estimated per index granule but the bloom filter exists per segment.
```

Comment:
Thinking about it, the sizing is better now (thanks), but I think we still need these two enhancements to make it perfect (--> fine to keep these TODOs around for now).